src/test/java/com/fasterxml/jackson/databind/ser/SimpleFilterProviderTest.java

```diff
@@ -0,0 +1,138 @@
+
+        assertEquals(a2q("{'a':'1a','b':'2b'}"), jsonString);
+    }
+
+    public void testAddFilterNulls() throws Exception {
+        FilterProvider prov = new SimpleFilterProvider()
+                .addFilter("filterB", SimpleBeanPropertyFilter.filterOutAllExcept(null, null));
+        AnyBeanB beanB = new AnyBeanB("1a", "2b");
+
+        String jsonString = MAPPER.writer(prov).writeValueAsString(beanB);
+
+        assertEquals(a2q("{}"), jsonString);
+    }
+
+    public void testAddFilterEmptyString() throws Exception {
+        FilterProvider prov = new SimpleFilterProvider()
+                .addFilter("filterB", SimpleBeanPropertyFilter.filterOutAllExcept("", ""));
+        AnyBeanB anyBeanB = new AnyBeanB("1a", "2b");
+
+        String jsonString = MAPPER.writer(prov).writeValueAsString(anyBeanB);
+
+        assertEquals(a2q("{}"), jsonString);
+    }
+
+    @JsonFilter(value = "")
+    public static class AnyBeanC
+    {
+        public String c;
+        public String d;
+
+        public AnyBeanC(String c, String d) {
+            this.c = c;
+            this.d = d;
+        }
+    }
+
+    public void testAddFilterWithEmptyStringId() throws Exception {
+        FilterProvider prov = new SimpleFilterProvider()
+                .addFilter("", SimpleBeanPropertyFilter.filterOutAllExcept("d"));
+        AnyBeanC bean = new AnyBeanC(null, "D is filtered");
+
+        String jsonString = MAPPER.writer(prov).writeValueAsString(bean);
+
+        assertEquals(a2q("{'c':null,'d':'D is filtered'}"), jsonString);
+    }
+
+    @JsonFilter("filterD")
+    public static class AnyBeanD
+    {
+        public String a;
+        public String b;
+
+        public AnyBeanD(String a, String b) {
+            this.a = a;
+            this.b = b;
+        }
+    }
+
+    public void testAddingNullFilter2ThrowsException() {
+        FilterProvider prov = new SimpleFilterProvider()
+                .addFilter("filterD", null);
+        ObjectWriter writer = MAPPER.writer(prov);
+        AnyBeanD beanD = new AnyBeanD("1a", "2b");
+
+        try {
+            writer.writeValueAsString(beanD);
+            fail("Should not have passed");
+        } catch (JsonMappingException e) {
+            verifyException(e, "No filter configured with id 'filterD'");
+        } catch (Exception e) {
+            fail("Should not have passed");
+        }
+    }
+
+    public void testAddingNullFilterIdThrowsException() {
+        FilterProvider prov = new SimpleFilterProvider()
+                .addFilter(null, SimpleBeanPropertyFilter.filterOutAllExcept("a", "b"));
+        ObjectWriter writer = MAPPER.writer(prov);
+        AnyBeanD beanD = new AnyBeanD("1a", "2b");
+
+        try {
+            writer.writeValueAsString(beanD);
+            fail("Should not have passed");
+        } catch (JsonMappingException e) {
+            verifyException(e, "No filter configured with id 'filterD'");
+        } catch (Exception e) {
```

Comment:
Just remove this, no need to catch (it doesn't pass if exception thrown but fails)